API-884 Extract health APIs parent

jblefkove-va opened the pull request:
https://vasdvp.atlassian.net/browse/API-884

- Extract `health-apis-parent` pom, `api-starter`, `sentinel` core, `service-auto-config`, and `service-starter`. New version is `1.1.0-SNAPSHOT`. 
- New `sentinel` library has data-query-agnostic classes. (Simple library that just builds a `jar`. No integration tests or docker magic.)
- Change sentinel package from `gov.va.health.api.*` to `gov.va.api.health.*`.

(This does not include `test-starter` pom or the associated docker machinery.)

bschofield-va commented on README.md:
```diff
@@ -1 +1,43 @@
+git-secrets must be installed and configured to scan for AWS entries and the patterns in
+[.git-secrets-patterns](.git-secrets-patterns). Exclusions are managed in 
+[.gitallowed](.gitallowed).
+The [init-git-secrets.sh](src/scripts/init-git-secrets.sh) script can be used to simply set up.
```
lies!

bschofield-va replied:
Git secrets can be enabled in this repo using
```
git secrets --register-aws
git secrets --add-provider -- cat .git-secrets-patterns
```

jblefkove-va replied:
Updated!

bschofield-va commented:
Looking through Sentinel...

- Categories `Local` and `Manual` are meaningful across products, but `*Argo/Cargo` are not. Ultimately, they should be something like `ProdDataQueryPatient` and `ProdDataQueryClinician`.
- I don't really understand why `SystemDefinition` is needed. `SystemDefinitions` in `health-apis` has a useful mechanisms to override values in `ServiceDefinition` based on system properties. Perhaps those could be brought forward.
- I wonder if `IdMeOauthRobot` and `DevPortal` should be packaged together separately. They don't really fit with the rest of the classes in `...health.sentinel`

jblefkove-va commented:
* I moved the patient and clinician categories to `data-query-tests` and renamed them.
* Removed `SystemDefinition`.
* Moved `IdMeOauthRobot` and `DevPortal` to `selenium` package.

I am still mulling over how to define `ServiceDefinition`s in terms of system properties.

jblefkove-va commented:
* I moved the patient and clinician categories to `data-query-tests` and renamed them.
* Removed `SystemDefinition`.
* Moved `IdMeOauthRobot` and `DevPortal` to `selenium` package.

I am still mulling over how to define `ServiceDefinition`s in terms of system properties.

@bschofield-va

jblefkove-va commented:
@bschofield-va I pulled the `optionUrl()` and `optionApiPath()` methods into a `SentinelProperties` class and did some refactoring in `SystemDefinitions` to reduce the duplication.
